lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/DelegateDecoders.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader.decoders;
+
+import io.trino.parquet.reader.SimpleSliceInputStream;
+
+import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_MICROSECOND;
+import static java.util.Objects.requireNonNull;
+
+public class DelegateDecoders
+{
+    private DelegateDecoders() {}
+
+    public static ValueDecoder<long[]> timeMicrosDecoder(ValueDecoder<long[]> delegate)
```

Comment:
This is only used in the `getTimeMicrosDecoder` method. No point in creating a generic method for this, so just inline the logic there instead.

Reply:
I've removed `DelegateDecoder` interface, but I'd prefer to keep the definition of the ValueDecoder here rather than inlining it into `ValueDecoders#getTimeMicrosDecoder` because `ValueDecoders` is cleaner by keeping ValueDecoder definitions out of it. We'll also be adding more such implementations to `DelegateDecoders` in future PRs.

Reply:
"DelegateDecoders" is a weird concept. To me, they are just an implementation detail for specific value decoders, so the split into a separate class feels very artificial.

Reply:
I would like to keep all value decoder implementations out of `ValueDecoders`, we should have just mappings from `ParquetEncoding` to `ValueDecoder` implementation there.
The idea behind not writing this in `ApacheParquetValueDecoder` directly and composing it on top of `ValueDecoder<long[]>` is that in future PRs the parquet-mr decoders would be swapped out by a decoder written inside trino. This allows that swap to take place more easily without repeating the code to convert raw long into picosecond values for the time(6) type. There will be other cases that can be handled this way as well.